system/Cache/Handlers/BaseHandler.php

```diff
@@ -14,12 +14,65 @@
 use Closure;
 use CodeIgniter\Cache\CacheInterface;
 use Exception;
+use InvalidArgumentException;
 
 /**
  * Base class for cache handling
  */
 abstract class BaseHandler implements CacheInterface
 {
+	/**
+	 * Reserved characters that cannot be used in a key or tag.
+	 *
+	 * @see https://github.com/symfony/cache-contracts/blob/c0446463729b89dd4fa62e9aeecc80287323615d/ItemInterface.php#L43
+	 */
+	public const RESERVED_CHARACTERS = '{}()/\@:';
+
+	/**
+	 * Maximum key length.
+	 */
+	public const MAX_KEY_LENGTH = PHP_INT_MAX;
+
+	/**
+	 * Prefix to apply to cache keys.
+	 * May not be used by all handlers.
+	 *
+	 * @var string
+	 */
+	protected $prefix;
+
+	/**
+	 * Validates a cache key according to PSR-6.
+	 * Keys that exceed MAX_KEY_LENGTH are hashed.
+	 *
+	 * @param string $key    The key to validate
+	 * @param string $prefix Optional prefix to include in length calculations
+	 *
+	 * @throws InvalidArgumentException When $key is not valid
+	 *
+	 * @see https://github.com/symfony/cache/blob/7b024c6726af21fd4984ac8d1eae2b9f3d90de88/CacheItem.php#L158
+	 */
+	public static function validateKey($key, $prefix = ''): string
+	{
+		if (! is_string($key))
+		{
+			throw new InvalidArgumentException('Cache key must be a string');
+		}
+		if ($key === '')
+		{
+			throw new InvalidArgumentException('Cache key cannot be empty.');
+		}
+		if (strpbrk($key, self::RESERVED_CHARACTERS) !== false)
+		{
+			throw new InvalidArgumentException('Cache key contains reserved characters ' . self::RESERVED_CHARACTERS);
+		}
+
+		// If the key with prefix exceeds the length then return the hashed version
+		return strlen($prefix . $key) > static::MAX_KEY_LENGTH ? md5($key) : $key;
```

Comment:
Why is the `$prefix` not included in the return?

Reply:
Two-fold: 1) We have no specification around prefix handling, so there is no assumption that handlers will use them even when set (even our internal handlers are inconsistent), and 2) one of the goals of this method is to keep it "generic" enough to be used by the PSR-6 adapters as well, which will probably ignore this return value (so as not to double-process keys) but definitely won't deal in prefixes at all.

Reply:
What's your take on making this a `void` return type, i.e., when the length of the key exceeds the max length it throws an `InvalidArgumentException` too just like the previous ifs?

I am thinking why is there a need to use MD5 hashes of keys. Since an excessively long key is an "invalid" key, then should it follow this:
> * Any time an invalid argument is passed into a method it must throw an
> * exception class which implements Psr\Cache\InvalidArgumentException.

Reply:
I'm taking my lead from the PHP-Cache tests since that is what I plan on using to test the PSR-6 & 16 adapters. According to the test case (which is not FIG-official mind you) excessively long keys *should* be acceptable. It's the last dialing test on my implementation which is what instigated this PR.

Reply:
Okay, let's proceed then. I've looked at your branch and the code looks good so I'll trust you on this.

Reply:
Awesome, thanks for checking it out! I'll fix the prefixes and run through tests again.

Reply:
Actually after working with this some more I think not returning the prefix is pretty confusing and outweighs the benefits. Since the PSR-6 adapter will ignore the return value I think I will go with your suggestion.